doc/classes/PhysicsServer2D.xml

```diff
@@ -16,158 +17,169 @@
+				Returns the [RID] of the shape with the given index in the area's array of shapes.
 			</description>
 		</method>
 		<method name="area_get_shape_count" qualifiers="const">
 			<return type="int" />
 			<param index="0" name="area" type="RID" />
 			<description>
-				Returns the number of shapes assigned to an area.
+				Returns the number of shapes added to the area.
 			</description>
 		</method>
 		<method name="area_get_shape_transform" qualifiers="const">
 			<return type="Transform2D" />
 			<param index="0" name="area" type="RID" />
 			<param index="1" name="shape_idx" type="int" />
 			<description>
-				Returns the transform matrix of a shape within an area.
+				Returns the local transform matrix of the shape with the given index in the area's array of shapes.
 			</description>
 		</method>
 		<method name="area_get_space" qualifiers="const">
 			<return type="RID" />
 			<param index="0" name="area" type="RID" />
 			<description>
-				Returns the space assigned to the area.
+				Returns the [RID] of the space assigned to the area. Returns [code]RID()[/code] if no space is assigned.
 			</description>
 		</method>
 		<method name="area_get_transform" qualifiers="const">
 			<return type="Transform2D" />
 			<param index="0" name="area" type="RID" />
 			<description>
-				Returns the transform matrix for an area.
+				Returns the transform matrix of the area.
 			</description>
 		</method>
 		<method name="area_remove_shape">
 			<return type="void" />
 			<param index="0" name="area" type="RID" />
 			<param index="1" name="shape_idx" type="int" />
 			<description>
-				Removes a shape from an area. It does not delete the shape, so it can be reassigned later.
+				Removes the shape with the given index from the area's array of shapes. The shape itself is not deleted, so it can continue to be used elsewhere or added back later. As a result of this operation, the area's shapes which used to have indices higher than [param shape_idx] will have their index decreased by one.
 			</description>
 		</method>
 		<method name="area_set_area_monitor_callback">
 			<return type="void" />
 			<param index="0" name="area" type="RID" />
 			<param index="1" name="callback" type="Callable" />
 			<description>
+				Sets the area's area monitor callback. This callback will be called when any other (shape of an) area enters or exits (a shape of) the given area, and must take the following five parameters:
+				1. an integer [code]status[/code]: either [constant AREA_BODY_ADDED] or [constant AREA_BODY_REMOVED] depending on whether the other area's shape entered or exited the area,
+				2. an [RID] [code]area_rid[/code]: the [RID] of the other area that entered or exited the area,
+				3. an integer [code]instance_id[/code]: the [code]ObjectID[/code] attached to the other area,
+				4. an integer [code]area_shape_idx[/code]: the index of the shape of the other area that entered or exited the area,
```

Comment:
what about  something like `over_area_shape_idx` to avoid confusion when reading quickly ?

Reply:
I've kept it as is for now, I think it's alright.